fix: concerto-linter tests fail on fresh clones

# Bug Report 
Tests fail in concerto-linter package when running npm test on a fresh clone or clean environment due to missing build dependency.
## Expected Behavior
Running npm test in the concerto-linter package should build all necessary dependencies and run tests successfully.
## Current Behavior
Tests fail with module resolution error: src/index.ts:19:31 - error TS2307: Cannot find module '@accordproject/concerto-linter-default-ruleset'
## Possible Solution
Update the pretest script in concerto-linter/package.json to build the default-ruleset workspace before running tests: "pretest": "npm run build -w @accordproject/concerto-linter-default-ruleset && npm-run-all lint"
## Steps to Reproduce
1. Fresh clone the repository: git clone https://github.com/accordproject/concerto.git
2. Install dependencies: npm install
3. Navigate to linter package: cd packages/concerto-linter
4. Run tests: npm test
## Context (Environment)
### Desktop
 - OS: Linux/macOS/Windows (affects all)
 - Browser: N/A
 - Version: 3.25.0
## Detailed Description
The concerto-linter package depends on @accordproject/concerto-linter-default-ruleset, which is a local workspace located at packages/concerto-linter/default-ruleset. The import statement on line 19 of src/index.ts tries to import from this package: import concertoRuleset from '@accordproject/concerto-linter-default-ruleset'. However, the dist/ folder of default-ruleset doesn't exist until it's built. The current pretest script only runs lint, not build: "pretest": "npm-run-all lint". This causes TypeScript to fail when trying to resolve the module, since it looks for dist/ruleset-main.js as specified in the default-ruleset package.json main field.
## Possible Implementation
The fix is simple - ensure default-ruleset is built before tests run: "pretest": "npm run build -w @accordproject/concerto-linter-default-ruleset && npm-run-all lint". This uses npm workspaces (-w flag) to build just the default-ruleset package before running lint and tests.

--- a/DEVELOPERS.md
+++ b/DEVELOPERS.md
@@ -23,6 +23,30 @@ git remote add upstream "https://github.com/accordproject/concerto.git"
 
 # Install node.js dependencies:
 npm i
+
+# Build all packages:
+npm run build
+```
+
+#### Running Tests
+
+Before running tests, you must first build the project to compile all workspace dependencies:
+
+```shell
+# Build all packages (required before testing):
+npm run build
+
+# Run all tests:
+npm run test
+```
+
+> **Note:** The build step is required because some packages (like `concerto-linter`) depend on other workspace packages that need to be compiled first. The CI workflows follow this same pattern: build first, then test.
+
+If you're working on a specific package, you can run tests for just that package:
+
+```shell
+# Run tests for a specific package:
+npm run test -w @accordproject/concerto-core
 ```
 
 [apdev]: https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md

